### PR TITLE
Coordinators + iPad support

### PR DESCRIPTION
**Pull request by xmollv**
This PR finishes #23 and it's heavily inspired on the PR that @atineoSE opened a while ago. I went down a rabbit hole and did the iPad support + Coordinators everything on the same branch 🤷🏼‍♂️

I'm pushing this PR but should not be merged yet, I want to review it tomorrow when I'm not as tired as I am right now.

### CODE REVIEW

File: llitgi/Coordinators/AppCoordinator.swift [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    init(window: UIWindow, tabBarController: UITabBarController = UITabBarController(), factory: ViewControllerFactory, userManager: UserManager) {
+        self.splitViewController = UISplitViewController()
+        self.tabBarController = tabBarController
+        self.emptyViewController = factory.instantiateEmptyDetail()
+        self.factory = factory
+        self.userManager = userManager
+        
+        super.init()
+        
+        let listViewController: ListViewController = self.factory.instantiateList(for: .myList)
```
Comment (xmollv) [inline llitgi/Coordinators/AppCoordinator.swift]: This code to configure the Lists depending on the type should be hidden elsewhere.

**Comment by xmollv** [comment]
@atineoSE It would be awesome if you could give a try to this code to see how it behaves on a real device and if you have any input/suggestion, I'd love to hear it and improve the PR. I want to be very cautious with this, because when you support iPad, there's no going back on the App Store...

**Comment by atineoSE** [comment]
@xmollv alright, I'll give it a spin with a real iPad and let you know. It may take me a few days though.

**Comment by xmollv** [comment]
@atineoSE sure, no pressure whatsoever. I still want to tweak a few things on this PR, but I want to make sure that on a real device there are no issues! 👍

**Comment by xmollv** [comment]
I had to get rid of the Empty state for the detail (even tough I think that it looked better with it) because the UISplitViewController doesn't behave as expected with it. I tried to subclass UINavigationController (the one used for the detail) and override `loadView` to add the simple empty state, but nope, UISplitViewController is not cooperating at all.

**Comment by atineoSE** [comment]
@xmollv I tested on a real iPad (iPad Pro (12.9-inch) running iOS 11.4.1 - current version as of this writing) and it works beautifully. I tried Split View and Slide Over both in landscape and portrait and it all works well. 

I went through all major functions and there was only one issue, possibly a regression: when tapping on the plus sign (+) to add a new article, and provided there is nothing in the pasteboard, it shows a spinner for about 1 second and nothing happens. In the App Store version there is an alert to inform the user that the pasteboard is empty.

**Comment by xmollv** [comment]
@atineoSE Thanks! I'll double check the adding behaviour to see if I broke something.

I'm also leaving a reminder to myself about some bugs that i found:
- The login screen on a Plus device in landscape is broken. We need to do some magics with horizontalSizeClass == .regular && verticalSizeClass == .compact to avoid it.
- The settings screen (or any other modal) is dismissed if we switch from a compact environment to a regular one. This happens because of the fix to dismiss the SFS when this behaviour occurs, but the other VC's should not be dismissed.

**Pull request merged**